### DISCUSSION TITLE
Using websocket with SSL for foxglove

### DISCUSSION THREAD

**Discussion by Zeki411** [foxglove/community#1008] [Q&A]
Hi, everyone,

I'm trying to set up Foxglove Bridge with WebSocket SSL. I've used Let's Encrypt with Certbot to generate the certificate and key, and I've added them to the launch commands as instructed.

Here is the command I used:

> ros2 launch foxglove_bridge foxglove_bridge_launch.xml   tls:=true   certfile:=/etc/letsencrypt/live/mydomain/cert.pem   keyfile:=/etc/letsencrypt/live/mydomain/privkey.pem

I used DuckDNS to create my domain.

However, when I try to connect from both the Foxglove web app and Foxglove Studio to wss://localhost:8765, it fails, and Foxglove Bridge prints out the following error:

> [foxglove_bridge-1] [ERROR] [1719055657.592036654] [foxglove_bridge]: [WS] handle_transport_init received error: TLS handshake failed
> [foxglove_bridge-1] [INFO] [1719055657.592215377] [foxglove_bridge]: [WS] asio async_shutdown error: asio.ssl:167772567 (shutdown while in init)

Can anyone help me with this issue?

Thank you!

**Comment by achim-k** [comment]
Does it work when you connect using your domain name `wss://<domain>.duckdns.org:8765` ?
